src/scalarstats.jl

```diff
@@ -197,75 +194,117 @@ nquantile(v::AbstractArray{<:Real}, n::Integer) = quantile(v, (0:n)/n)
 """
     span(x)
 
-Return the span of an integer array, i.e. the range `minimum(x):maximum(x)`.
-The minimum and maximum of `x` are computed in one-pass using `extrema`.
+Return the span of a collection, i.e. the range `minimum(x):maximum(x)`.
+The minimum and maximum of `x` are computed in one pass using `extrema`.
 """
-span(x::AbstractArray{<:Integer}) = ((a, b) = extrema(x); a:b)
+span(x) = ((a, b) = extrema(x); a:b)
 
 # Variation coefficient: std / mean
 """
     variation(x, m=mean(x))
 
-Return the coefficient of variation of an array `x`, optionally specifying
+Return the coefficient of variation of collection `x`, optionally specifying
 a precomputed mean `m`. The coefficient of variation is the ratio of the
 standard deviation to the mean.
 """
-variation(x::AbstractArray{<:Real}, m::Real) = stdm(x, m) / m
-variation(x::AbstractArray{<:Real}) = variation(x, mean(x))
+variation(x, m) = stdm(x, m) / m
+variation(x) = ((m, s) = mean_and_std(x); s/m)
+
+# Standard error of the mean: std / sqrt(len)
+# Code taken from var in the Statistics stdlib module
+
+# faster computation of real(conj(x)*y)
+realXcY(x::Real, y::Real) = x*y
+realXcY(x::Complex, y::Complex) = real(x)*real(y) + imag(x)*imag(y)
 
-# Standard error of the mean: std(a) / sqrt(len)
 """
-    sem(a)
+    sem(x)
 
-Return the standard error of the mean of `a`, i.e. `sqrt(var(a) / length(a))`.
+Return the standard error of the mean of collection `x`,
+i.e. `sqrt(var(x, corrected=true) / length(x))`.
 """
-sem(a::AbstractArray{<:Real}) = sqrt(var(a) / length(a))
+function sem(x)
+    y = iterate(x)
+    if y === nothing
+        T = eltype(x)
+        return oftype(sqrt((abs2(zero(T)) + abs2(zero(T)))/2), NaN)
```

Comment:
This is mildly insane,
but I see why you have to do it.

Probably should comment on it.

Noting that

```
julia> T = Complex
Complex

julia> convert(T, NaN)
NaN + 0.0im

julia> oftype(sqrt((abs2(zero(T)) + abs2(zero(T)))/2), NaN)
NaN
```

Reply:
Note that this is a slightly adaption from `Base.var`, so I'd rather not change it.

Reply:
I am not suggesting changing it.
But explaining it might be good.
Even if it is just 
`Return the Nan of the type that we would get, had this collection contained any elements`